packages/react/src/components/CommandCenter/SuggestionsStore.ts

```diff
@@ -0,0 +1,67 @@
+import { IReactionDisposer, makeAutoObservable, reaction } from "mobx";
+import { isEqual } from "lodash";
+import { Suggestion } from "@/constants/commandTypes";
+
+
+export class SuggestionsStore {
+    private _suggestions: Suggestion[] = [];
+    private _highlightedSuggestion?: Suggestion;
+    private _disposers: IReactionDisposer[] = [];
+
+    constructor() {
+        makeAutoObservable(this);
+    }
+
+    get suggestions() {
+        return this._suggestions;
+    }
+
+    set suggestions(suggestions: Suggestion[]) {
+        this._suggestions = suggestions;
+    }
+
+    get highlightedSuggestion() {
+        return this._highlightedSuggestion;
+    }
+
+    set highlightedSuggestion(suggestion: Suggestion | undefined) {
+        this._highlightedSuggestion = suggestion;
+    }
+
+    highlightNextSuggestion() {
+        if (!this.suggestions.length) return;
+        const currentIndex = this.suggestions.findIndex(s => s === this._highlightedSuggestion);
+        const nextIndex = (currentIndex + 1) % this.suggestions.length;
+        this._highlightedSuggestion = this.suggestions[nextIndex];
+    }
+
+    highlightPreviousSuggestion() {
+        if (!this.suggestions.length) return;
+        const currentIndex = this.suggestions.findIndex(s => s === this._highlightedSuggestion);
+        const previousIndex = (currentIndex - 1 + this.suggestions.length) % this.suggestions.length;
+        this._highlightedSuggestion = this.suggestions[previousIndex];
+    }
```

Comment:
We do expensive find operations each time, but it's deemed acceptable as we are expected to have a small number of suggestions.